src/programacion/empresaprogramacion/Empresa.java

```diff
@@ -1,54 +1,216 @@
 package programacion.empresaprogramacion;
+
+import programacion.empresaprogramacion.excepciones.ParametroInvalidoException;
+import programacion.empresaprogramacion.excepciones.RutaArchivoInvalida;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
+import java.util.TreeMap;
 
 public class Empresa {
-    
-    public String nombre;
-    public Map<Integer, Empleado> mapaEmpleados;
 
-    public Empresa(String nombre) {
-        this.nombre = nombre;
-        //TODO this.empleados = new ?<>();
-    }
+  public String nombre;
+  public Map<String, Empleado> mapaEmpleados;
+  static final String CSV_FILENAME_PROGRAMADORES = "programadores.csv";
+  static final String CSV_FILENAME_GERENTES = "gerentes.csv";
 
-    public void addEmpleado(Empleado empleado) {
-        // TODO  (no permitir meter 2 empleados con el mismo "idEmpleado" y dado el caso lanzar 
-        //exception)
+  public Empresa(String nombre) {
+    this.nombre = nombre;
+    mapaEmpleados = new HashMap<>();
+  }
+
+  public void addEmpleado(Empleado empleado) throws ParametroInvalidoException {
+    if (mapaEmpleados.containsKey(empleado.DNI)) {
+      if (empleado.tipoEmpleado == TipoEmpleado.PROGRAMADOR) {
+        throw new ParametroInvalidoException("Ya existe un Programador con ese ID de empleado.\n");
```

Comment:
Por decir algo, podrías ahorrarte el if/else:

```suggestion
        throw new ParametroInvalidoException("Ya existe un " + empleado.tipoEmpleado + " con ese ID de empleado.\n");
```